prow/plugins/trigger/generic-comment_test.go

```diff
@@ -1481,3 +1483,63 @@ func TestRetestFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestFailingWorkflow(t *testing.T) {
+
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			t.Errorf("Bad method: %s", r.Method)
+		}
+		accept := r.Header["Accept"]
+		if accept[0] != "application/vnd.github.v3+json" {
+			t.Errorf("Bad request accept: %s", accept)
+		}
+
+		options := strings.Split(r.URL.Path, "/")
+		runID := options[6]
+
+		switch runID {
+		case "1":
+			w.WriteHeader(http.StatusForbidden)
+		case "2":
+			http.Error(w, `{"message": "Unable to retry this workflow run because it was created over a month ago","documentation_url": "https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}`, http.StatusForbidden)
+		default:
+			w.WriteHeader(http.StatusCreated)
+			fmt.Fprint(w, "Re-run failed jobs from a workflow run - 201 Created")
+		}
+	}))
+	defer ts.Close() 
```

Comment:
Thinking about this a little more -- I don't think you need to create a test server here. It would be easier if you modified the `FakeClient` that we have, specifically this method:
https://github.com/kubernetes/test-infra/blob/44aa1dbdbca77ac9e45d00c0c32681d547e6c289/prow/github/fakegithub/fakegithub.go#L1196-L1198

You can add a new field to the `FakeClient` called `TriggerFailedGitHubWorkflowError` similar to:
https://github.com/kubernetes/test-infra/blob/44aa1dbdbca77ac9e45d00c0c32681d547e6c289/prow/github/fakegithub/fakegithub.go#L142

And you can set this field each time you create a new FakeClient in each subtest, and based on if you're setting the error or not, you can return the desired error message in the fake implementation.

This way you also won't have to modify the `github.NewFakeClient()` method.

Reply:
Hey I did write the test here is how it looks right now 
```
func TestRandom(t *testing.T) {
	var testCases = []struct {
		name            string
		triggerFailedGitHubWorkflowError error 
		runID           int
		expectedMessage string
	}{
		{
			name:            "Triggered workflow failed without err",
			runID:           1,
			triggerFailedGitHubWorkflowError: errors.New(``),
			expectedMessage: "",
		},
		{
			name:  "Triggered workflow failed due to expired workflow",
			triggerFailedGitHubWorkflowError: errors.New(`{"message": "Unable to retry this workflow run because it was created over a month ago","documentation_url": "https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}`),
			runID: 2,
			expectedMessage: `{"message": "Unable to retry this workflow run because it was created over a month ago","documentation_url": "https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}`,
		},
	}
	c := fakegithub.NewFakeClient()
	for _, testCase := range testCases {
		t.Run(testCase.name, func (t *testing.T) {
			c.TriggerFailedGitHubWorkflowError = testCase.triggerFailedGitHubWorkflowError
			err := c.TriggerFailedGitHubWorkflow("org", "repo", testCase.runID)
			if err != nil {
				if !strings.Contains(err.Error(), testCase.expectedMessage) {
					t.Errorf("Expected error message:\n%s\nGot:\n%s", testCase.expectedMessage, err.Error())
				}
			} else {
				if testCase.expectedMessage != "" {
					t.Errorf("Expected error message:\n%s\nGot: %s", testCase.expectedMessage, err)
				}
			} 
		})
	}
}
```

Reply:
Go for it and update the PR! Its easier for me to leave comments once you do that. But your general approach looks good and is also what I had in mind 👍🏼